docs/codeql/codeql-language-guides/customizing-library-models-for-ruby.rst

```diff
@@ -0,0 +1,406 @@
+- **Argument[any-named]** selects any keyword argument.
+- **Argument[hash-splat]** selects a special argument representing all keyword arguments passed in the method call.
+- **Parameter[**\ `number`\ **]** selects the argument at the given index.
+- **Parameter[**\ `string`:\ **]** selects the keyword argument with the given name.
+- **Parameter[self]** selects the **self** parameter of a method.
+- **Parameter[block]** selects the block parameter.
+- **Parameter[any]** selects any parameter, except self or block parameters.
+- **Parameter[any-named]** selects any keyword parameter.
+- **Parameter[hash-splat]** selects the hash splat parameter, often written as **\*\*kwargs**.
+- **ReturnValue** selects the return value of a call.
+- **Method[**\ `name`\ **]** selects a call to the method with the given name.
+- **Element[any]** selects any element of an array or hash.
+- **Element[**\ `number`\ **]** selects an array element at the given index.
+- **Element[**\ `string`\ **]** selects a hash element at the given key.
+- **Field[@**\ `string`\ **]** selects an instance variable with the given name.
+- **Fuzzy** selects all values that are derived from the current value through a combination of the other operations described in this list.
+  For example, this can be used to find all values that appear to originate from a particular class. This can be useful for finding method calls
+  from a known class, but where the receiver type is not known or is difficult to model.
+
+Additional notes about the syntax of operands:
+
+- Multiple operands may be given to a single component, as a shorthand for the union of the operands. For example, **Method[foo,bar]** matches the union of **Method[foo]** and **Method[bar]**.
+- Numeric operands to **Argument**, **Parameter**, and **Element** may be given as a lower bound. For example, **Argument[1..]** matches all arguments except 0.
+
+Kinds
+-----
+
+Source kinds
+~~~~~~~~~~~~
+
+- **remote**: A generic source of remote flow. Most taint-tracking queries will use such a source. Currently this is the only supported source kind.
+
+Sink kinds
+~~~~~~~~~~
+
+Unlike sources, sinks tend to be highly query-specific, rarely affecting more than one or two queries.
+Not every query supports customizable sinks. If the following sinks are not suitable for your use case, you should add a new query.
+
+- **code-injection**: A sink that can be used to inject code, such as in calls to **eval**.
+- **command-injection**: A sink that can be used to inject shell commands, such as in calls to **child_process.spawn**.
+- **path-injection**: A sink that can be used for path injection in a file system access, such as in calls to **fs.readFile**.
+- **sql-injection**: A sink that can be used for SQL injection, such as in a MySQL **query** call.
+- **url-redirection**: A sink that can be used to redirect the user to a malicious URL.
+- **log-injection**: A sink that can be used for log injection, such as in a **console.log** call.
```

Comment:
Feel free to ignore as I'm not an SME, but just checking `console.log` is definitely applicable to Ruby and this wasn't a missed copy edit from the JS docs.